src/core/components/fields/LabeledTextField.test.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * @vitest-environment jsdom
+ */
+
+import { expect, test } from "vitest"
+import { fireEvent, render, screen } from "test/utils"
+import { Form, FormProps } from "src/core/components/fields/Form"
+import { z } from "zod"
+import LabeledTextField from "./LabeledTextField"
+
+interface MyFormProps<S extends z.ZodType<any, any>> extends FormProps<S> {
+  name: string
+  placeholder: string
+  label: string
+  inputType?: "text" | "password" | "email" | "number" | "textarea"
+  onChangeCallback: (val) => void
+}
+
+const placeholder = "Container placeholder"
+const name = "input 1"
+const label = "input_label"
+
+function MyForm<S extends z.ZodType<any, any>>(props: MyFormProps<S>) {
+  const { name, placeholder, inputType, onChangeCallback, label, ...formProps } = props
+
+  // {/*  */}
+  return (
+    <div>
+      <Form<S> {...formProps}>
```

Comment:
```suggestion
      <Form {...formProps}>
```

Could you explain why `<S>` is present here? This seems unnecessary but you may have a reason I am missing.

Reply:
It is not necessary and I will remove it.